feat (criteria-Mitra-modal): to show detailed information

doohanas opened the pull request:
## Task 
- View Detail Kriteria [Mitra](https://airtable.com/appdN4AdnU8FNs94M/tblRtWbyq60Fx1m6v/viwlpGjUkaHCytKMk/recNs0nqIo38NsrpB?blocks=hide)

## Changes
- Update` criteria-mitra modal` style and content
- Update `competition page`  to add `criteria-mitra` component
- Update default `layout `to add `modal-criteria-mitra`
- Update function button in `competition-criteri-mitra`

## Preview
1. Show modal criteria-mitra
![modal-criteria-mitra](https://user-images.githubusercontent.com/79241754/171111948-50fee2ff-53c0-46b1-bbab-e734d40e862e.gif)

## Evidence
title: feat (criteria-Mitra-modal): to show detailed information
project: Desa Digital
participants: @doohanas @yoslie @Ibwedagama

Ibwedagama commented on components/Competition/Criteria/Mitra/index.vue:
```diff
@@ -73,7 +73,7 @@ export default {
   methods: {
     onShowDetailContent () {
-      // @todo: to show info detail content
+      this.$emit('onClick', true)
```
minor:

hello @doohanas , according to official Vue.js docs, we should use `kebab-case` naming convention for custom events

ref: https://eslint.vuejs.org/rules/custom-event-name-casing.html

doohanas replied:
Hello kak @Ibwedagama this changes for custom event naming in [here](https://github.com/jabardigitalservice/dedi-landing-page/pull/129/commits/908af654335cb9996de73867fd87912b3ba25817)

Ibwedagama commented on components/CriteriaMitra/index.vue:
```diff
@@ -0,0 +1,214 @@
+                  <li
+                    v-for="(textDetail, index) in item.textDetails"
+                    :key="index"
+                    class="modal-criteria-mitra__container-content-main-text-info-message"
```
hello @doohanas,

this class name is very long and kinda hard to understand, is it the correct BEM convention? honestly, I'm not really know much about BEM 🙏🏼

doohanas replied:
Hello, kak @Ibwedagama,

I create this class based on BEM Convention. So, the BEM Convention is divided into three parts. 
1. Block 
Block is the top level of the component. This block should be thought of as a parent. In this case, my block is `modal-criteria-mitra`
2. Element
An element can be called a child element. In this case, my element is `container`. Then, the container has some children. So, We can add a dash or `-` to define a specific element.
3. Modifier
The modifier is to manipulate the block so that we can theme or style that particular component. We can add two dashes or `--` to define the modifier. 
The naming is too long because i have to add specific css for each element kak. 

The references: 
1. [BEM](http://getbem.com/naming/) Convention
2. BEM Convention from [css-trick.com](https://css-tricks.com/bem-101/)

Ibwedagama commented on components/CriteriaMitra/index.vue:
```diff
@@ -0,0 +1,214 @@
+              class="modal-criteria-mitra__container-content-main"
+            >
+              <div class="modal-criteria-mitra__container-content-main-img">
+                <img class="mx-auto" width="40px" height="40px" :src="item.image" :alt="item.textImg">
```
minor:

hello @doohanas I think we doesn't have to add `px` to `width` and `height` attribute on an <img> tag, cmiiw

doohanas replied:
😆 You're right the `width `and `height` attrs doesn't need `unit`

Ibwedagama commented on components/CriteriaMitra/index.vue:
```diff
@@ -0,0 +1,214 @@
+  methods: {
+    onClose (closeOn) {
+      if (closeOn !== 'overlay' || window.innerWidth < 640) {
+        this.$emit('onCloseModal', false)
```
minor:

same as the previous discussion 👍🏼

doohanas replied:
This changes same with the first [comment](https://github.com/jabardigitalservice/dedi-landing-page/pull/129/#discussion_r887576391)

yoslie commented:
<img width="732" alt="image" src="https://user-images.githubusercontent.com/65396086/171328922-42329b5a-3f8a-43e9-8441-a720473c2653.png">

please check again the handover section in figma style:
- check margin / padding
<img width="405" alt="image" src="https://user-images.githubusercontent.com/65396086/171329068-ab1880c2-c67b-49d3-b39c-dadaaa155026.png">

- check background color bottom
<img width="578" alt="image" src="https://user-images.githubusercontent.com/65396086/171329044-74d6b2c6-9069-48b9-bac0-e96a951c6dda.png">

btw I resolve conflict, so please pull from origin feat/view-detail-mitra to your local feat/view-detail-mitra first 🙏

doohanas commented:
Hello kak @yoslie, Based on your comment [here](https://github.com/jabardigitalservice/dedi-landing-page/pull/129/#issuecomment-1143110691)

I fixed some issues:
1. [Padding ](https://github.com/jabardigitalservice/dedi-landing-page/pull/129/commits/91132f2b3d7bb55c2de0096b85e5503b0d9d4de4) modal-criteria-mitra
2. [Button ](https://github.com/jabardigitalservice/dedi-landing-page/pull/129/commits/7105630b9507c0584b03b9bacacc636f174c7ba7) size and padding
3. Added [padding](https://github.com/jabardigitalservice/dedi-landing-page/pull/129/commits/3895c423fc94ec1c7b97b1bd906accdb07ed107b) for modal
